### PR TITLE
Application factory

### PR DESCRIPTION
**Pull request by dhruvja**
i have created a initialize function where we have to do a new token mint.

### CODE REVIEW

File: programs/application_factory/src/lib.rs [hunk 1]
```diff
@@ -2,14 +2,50 @@ use anchor_lang::prelude::*;
...
 #[program]
 pub mod application_factory {
     use super::*;
 
-    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+    pub fn initialize(ctx: Context<Initialize>, job_ad_id: String) -> Result<()> {
+
+        let details = &mut ctx.accounts.base_account;
+
+        details.status = false,
```
Comment (blasrodri) [inline programs/application_factory/src/lib.rs]: Can you create a method for this object that's call something like "initialize" or "reset" and then we avoid having so many lines in the instruction by calling it directly?

File: programs/application_factory/src/lib.rs [hunk 2]
```diff
@@ -2,14 +2,50 @@ use anchor_lang::prelude::*;
...
+    #[account(init, payer = authority, seeds = [JOB_APPLICATION_SEED, job_ad_id.as_bytes()[..18].as_ref(), job_ad_id.as_bytes()[18..].as_ref(), authority.key().as_ref()], bump, space = )]
+    pub base_account: Account<'info, JobApplication>,
+    #[account(mut)]
+    pub authority: Signer<'info>,
+    pub system_program: Program<'info, System>
+}
+
+#[account]
+pub struct JobApplication {
+    pub status: bool,
```
Comment (blasrodri) [inline programs/application_factory/src/lib.rs]: This will probably be an enum. Since there might be more than two states

**Review by blasrodri — CHANGES_REQUESTED** [review]

File: programs/application_factory/src/lib.rs (hunk 2 above)
Comment (dhruvja) [reply programs/application_factory/src/lib.rs]: ok ill do that...

File: programs/application_factory/src/lib.rs (hunk 1 above)
Comment (dhruvja) [reply programs/application_factory/src/lib.rs]: ok ill check it out

File: programs/application_factory/src/lib.rs [hunk 3]
```diff
@@ -2,14 +2,69 @@ use anchor_lang::prelude::*;
...
+
         Ok(())
     }
 }
 
 #[derive(Accounts)]
-pub struct Initialize {}
+#[instruction(job_ad_id: String)]
+pub struct Initialize<'info> {
+    #[account(init, payer = authority, seeds = [JOB_APPLICATION_SEED, job_ad_id.as_bytes()[..18].as_ref(), job_ad_id.as_bytes()[18..].as_ref(), authority.key().as_ref()], bump, space = 1 + 4 + 32 + 40)]
```
Comment (blasrodri) [inline programs/application_factory/src/lib.rs:37]: You need to add 8 for the discriminant on top of what you have

File: programs/application_factory/src/lib.rs [hunk 4]
```diff
@@ -2,14 +2,69 @@ use anchor_lang::prelude::*;
...
+#[derive(Account)]
+#[instruction(bump: u8)]
+pub struct UpdateStatus<'info> {
+    #[account(mut, seeds = [JOB_APPLICATION_SEED, job_ad_id.as_bytes()[..18].as_ref(), job_ad_id.as_bytes()[18..].as_ref(), authority.key().as_ref()], bump)]
+    pub base_account: Account<'info, JobApplication>,
+}
+
+#[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize)]
+pub enum JobStatus {
+    selected,
```
Comment (blasrodri) [inline programs/application_factory/src/lib.rs]: In rust these should be capitalized

**Review by blasrodri — APPROVED** [review]

File: programs/application_factory/src/lib.rs (hunk 4 above)
Comment (dhruvja) [reply programs/application_factory/src/lib.rs]: what should be capitalized? enum values or the enum name?

Comment (blasrodri) [reply programs/application_factory/src/lib.rs]: The enum variants. Check enum docs of rust 

enum MyEnum {
VariantA,
VariantB
}

**Review by blasrodri — APPROVED** [review]

**Pull request merged**